lib/ex_hls/client/utils.ex

```diff
@@ -40,4 +41,14 @@ defmodule ExHLS.Client.Utils do
   def stream_format_to_media_type(%AAC{}), do: :audio
   def stream_format_to_media_type(%RemoteStream{content_format: H264}), do: :video
   def stream_format_to_media_type(%RemoteStream{content_format: AAC}), do: :audio
+
+  @spec resolve_demuxing_engine_impl(String.t()) :: atom()
+  def resolve_demuxing_engine_impl(segment_uri) do
+    case Path.extname(segment_uri) do
```

Comment:
I think the more proper way to do it would be to do:
```
URI.parse!(segment_uri).path 
|> Path.extname()
|> case do
         ".ts" -> DemuxingEngine.MPEGTS
        ".m4s" -> DemuxingEngine.CMAF
        ".mp4" -> DemuxingEngine.CMAF
        _other -> raise "Unsupported segment URI extension: #{segment_uri |> inspect()}"
end
```
because this way we let `URI` retrieve the proper "path" for us, which we can later use with `Path.extname()`.